src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs

```diff
@@ -37,6 +40,29 @@ public override string Name
          }
       }
 
+      public void RemoveInitialCondition(InitialCondition initialCondition)
+      {
+         if (initialCondition == null)
+            return;
+         
+         _initialConditions.Remove(initialCondition.Path);
+         initialCondition.BuildingBlock = null;
+      }
+
+      public void AddInitialCondition(InitialCondition initialCondition)
+      {
+         _initialConditions.Add(initialCondition);
+         initialCondition.BuildingBlock = this;
+      }
+
+      public override void AcceptVisitor(IVisitor visitor)
+      {
+         base.AcceptVisitor(visitor);
+         _initialConditions.Each(ic => ic.AcceptVisitor(visitor));
+      }
+
+      public IEnumerable<InitialCondition> InitialConditions => _initialConditions;
```

Comment:
IReadonly?

Reply:
It's converting a Cache, so I don't know if it's better or not. I would need to call ToList on it for each access. This is templated from the PathAndValueEntityBuildingBlock which does the same.

Reply:
There is another thin. IReadonlyDictionary or sthg?

Reply:
![image](https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/assets/261477/f94c13f5-1e0e-4661-8a97-7f06ae88f815)

Looks like IReadOnlyCollection is the one.